cache.go

```diff
@@ -493,7 +502,7 @@ func (c *Cache[K, V]) Range(fn func(item *Item[K, V]) bool) {
 		i := item.Value.(*Item[K, V])
 		c.items.mu.RUnlock()
 
-		if !fn(i) {
+		if !i.isExpiredUnsafe() && !fn(i) {
```

Comment:
Since the mutex isn't locked here, the `isExpiredUnsafe()` shouldn't be used. What you can do is include the following before the unlock on line 503:
```go
expired := i.isExpiredUnsafe() // use in the if
```

Reply:
Fair enough, thanks. Updated